File: 2000-reverse-prefix-of-word/2000-reverse-prefix-of-word.cpp
class Solution {
public:
    string reversePrefix(string word, char ch) {
        int index = word.find(ch);
        if (index == -1) {
            return word;
        }
        
        string pre_reversed = word.substr(0, index + 1);
        reverse(pre_reversed.begin(), pre_reversed.end());

        return pre_reversed + word.substr(index+1);
    }
};

File: 2000-reverse-prefix-of-word/2000-reverse-prefix-of-word.py
class Solution:
    def reversePrefix(self, word: str, ch: str) -> str:
        index = word.find(ch)
        if index == -1:
            return word
        first_half = word[:index + 1]
        second_half = word[index + 1:]
        first_half_reverse = first_half[::-1]
        return first_half_reverse + second_half
        